Mathlib/Algebra/MvPolynomial/Equiv.lean

```diff
@@ -414,24 +414,49 @@ theorem optionEquivLeft_elim_eval (s : S₁ → R) (y : R) (f : MvPolynomial (Op
     optionEquivLeft_apply, Polynomial.map_X, Polynomial.eval_X, Option.elim_some, Polynomial.map_C,
     eval_X, Polynomial.eval_C, implies_true, and_self, φ]
 
+theorem support_coeff_optionEquivLeft {f : MvPolynomial (Option σ) R} {i : ℕ} {m : σ →₀ ℕ} :
+    m ∈ ((optionEquivLeft R σ f).coeff i).support ↔ m.optionElim i ∈ f.support := by
+  apply Iff.intro
+  · intro h
+    simpa [← optionEquivLeft_coeff_coeff] using h
+  · intro h
+    simpa [mem_support_iff, ← optionEquivLeft_coeff_coeff] using h
+
+lemma support_optionEquivLeft (p : MvPolynomial (Option σ) R) :
+    (optionEquivLeft R σ p).support
+      = Finset.image (fun m => m none) p.support := by
+  ext i
+  rw [Polynomial.mem_support_iff, Finset.mem_image, Finsupp.ne_iff]
+  constructor
+  · rintro ⟨m, hm⟩
+    refine ⟨optionElim i m, ?_, optionElim_apply_none _ _⟩
+    rw [← support_coeff_optionEquivLeft]
+    simpa using hm
+  · rintro ⟨m, h, rfl⟩
+    refine ⟨some m, ?_⟩
+    rwa [← coeff, zero_apply, ← mem_support_iff, support_coeff_optionEquivLeft, optionElim_some]
+
+theorem support_optionEquivLeft_nonempty {f : MvPolynomial (Option σ) R} (h : f ≠ 0) :
```

Comment:
I think the naming convention supports `nonempty_support_optionEquivLeft`

Reply:
There's also `support_finSuccEquiv_nonempty` in the file, so I will change that too to keep these analogous.

FWIW, I think the idea here is maybe that it reads like the informal phrase "the support of finSuccEquiv is nonempty". Search indicates that the string `_nonempty` appears 2835 times in Mathlib and `nonempty_` appears 2773 times, which suggests to me that this is an instance where the naming conventions are losing a battle with user intuition.